src/main/java/info/matai/services/server/ServerVersioningService.java

```diff
@@ -1,16 +1,128 @@
 package info.matai.services.server;
 
-import info.matai.models.enums.ServerTypes;
+import info.matai.App;
+import info.matai.models.configuration.Server;
+import info.matai.services.server.instances.PaperFetcherService;
+import org.springframework.util.StringUtils;
 
 public class ServerVersioningService {
-  public static String mcVersion;
-  public static ServerTypes serverType;
+  public static Server plugins;
+  public static Server pluginsLock;
 
-  public final String SERVER_JARS_API_URL = "";
+  public ServerVersioningService() {
+    this.initData();
+    IServerFetcherService fetcher = getFetcherService();
+    if (plugins != null) {
+      initWithPluginJson(fetcher);
+    } else if (pluginsLock != null) {
+      this.initWithPluginLockJson(fetcher);
+      // todo create plugins.yml file
+    } else {
+      // init
+    }
+  }
+
+  private IServerFetcherService getFetcherService() {
+    switch (ServerVersioningService.plugins.getServerType()) {
+      case PAPERMC:
+      case WATERFALL:
+      case VELOCITY:
+        return new PaperFetcherService(plugins.getServerType());
+      case SPIGOT:
+      // TODO add other fetchers
+      // return new SpigotFetcherService();
+
+      default:
+        return new PaperFetcherService(plugins.getServerType());
+    }
+  }
+
+  private void initWithPluginJson(IServerFetcherService fetcher) {
+    if (
+      StringUtils.hasText(ServerVersioningService.plugins.getMcVersion()) &&
+      StringUtils.hasText(ServerVersioningService.plugins.getServerBuild())
+    ) {
+      // Check if same as lock if not update
+      if (pluginsLock != null) {
+        // Check if plugin has same value as plugin-lock
+        if (
+          !ServerVersioningService
+            .plugins.getMcVersion()
+            .equals(ServerVersioningService.pluginsLock.getMcVersion()) ||
+          !ServerVersioningService
+            .plugins.getServerBuild()
+            .equals(ServerVersioningService.pluginsLock.getServerBuild())
+        ) {
+          App.LOGGER.info(
+            String.format(
+              "Downloading server version  %s-%s",
+              ServerVersioningService.plugins.getMcVersion(),
+              ServerVersioningService.plugins.getServerBuild()
+            )
+          );
+          fetcher.downLoadVersion(
+            ServerVersioningService.plugins.getMcVersion(),
+            ServerVersioningService.plugins.getServerBuild()
+          );
+          this.updateLock();
```

Comment:
Même comment qu'en au vis-à-vis le `this` et sans `this`, mais à voir que tu as le `this` pas mal partout, je le rajouterais en haut où ce qu'il l'a pas

Reply:
Me semble que mon intellisence me warn quand je met un this. Je vais rechecker demain.